mod/display.php

```diff
@@ -246,11 +246,11 @@ function display_content(App $a, $update = false, $update_uid = 0)
 	$page_uid = 0;
 
 	$parent = null;
-	if (!empty($parent_uri_id)) {
+	if (!local_user() && !empty($parent_uri_id)) {
 		$parent = Post::selectFirst(['uid'], ['uri-id' => $parent_uri_id, 'wall' => true]);
 	}
 
-	if (DBA::isResult($parent)) {
+	if (!local_user() && DBA::isResult($parent)) {
```

Comment:
The `!local_user()` condition here is superfluous since the one on line 249 already guarantees the `$parent` variable will keep the `null` value by never setting it to anything else on line 250.
```suggestion
	if (DBA::isResult($parent)) {
```